guardian/typerighter
# Setup rule manager infrastructure

## What does this change?

This PR sets up the necessary infrastructure for the new Rule Manager service. This includes a Web and DB stack.

The CDK is using the [@guardian/cdk ](https://github.com/guardian/cdk) package as a guinea pig project and also to ensure we stay aligned with future changes to how we handle our infrastrucutre. 

This change primarily sets up a pipeline to create the CloudFormation during CI and deploy them via riffRaff.

I'm happy to provide any additional detail to this description which would be useful. 

## How to test

This PR can be tested by ensuring that it run correctly in both the local and CODE environments. 
- For local this change should be a no-op and the rule-manager should continue to work as normal
- For CODE it should be possible to deploy the stacks and app and see the ELB healthchecks pass successfully.


## How can we measure success?

It is possible to deploy a working version of the Rule Manager and we have a CI and CD pipeline which enables us to iterate quickly. We are also easily able to make any necessary changes to our infrastructure through use of CDK.

## Have we considered potential risks?

During the development of this PR there was an issue found with the RDS password. Namely, the password parameter versions need to be inline with each other. Otherwise there will be issues deploying. This is likely to make changing the password frustrating in the future.  We might consider using some mapping for the password version number across environments, but the ideal solution would be to always use the latest (not currently possible with the `resolve` syntax).
